### PR TITLE
EVG-20029 Remove legacy UI project page and flag

### PR DESCRIPTION
**Pull request by hadjri**
EVG-20029

### Description
Now that the legacy project page has been deprecated, we should be free to remove the deprecation flag as well as the page itself.
### Testing
Tests should remain green.

### CODE REVIEW

File: service/project.go [hunk 1]
```diff
@@ -40,39 +40,8 @@ func (uis *UIServer) filterViewableProjects(u *user.DBUser) ([]model.ProjectRef,
...
-		NewUILink    string
-		SlackAppName string
-	}{allProjects, uis.GetCommonViewData(w, r, true, true), newUILink, slackAppName}
-
-	uis.render.WriteResponse(w, http.StatusOK, data, "base", "projects.html", "base_angular.html", "menu.html")
+	newUIProjectsLink := fmt.Sprintf("%s/projects", uis.Settings.Ui.UIv2Url)
+	http.Redirect(w, r, newUIProjectsLink, http.StatusPermanentRedirect)
 }
 
 func (uis *UIServer) projectPage(w http.ResponseWriter, r *http.Request) {
```
Comment (ablack12) [inline service/project.go:47]: Can we remove this too? And modifyProject?

Comment (hadjri) [reply service/project.go]: `projectPage` returns a JSON response of the project which is why I didn't include it as the removal, but I'm fine to get rid of it. Done.

**Review by ablack12 — COMMENTED** [review]
To test this, can you also deploy this to staging (if you didn't already) and verify that the waterfall, tasks, variants work as expected (and that the legacy project dropdown button still redirects to spruce)

**Comment by hadjri** [comment]
> To test this, can you also deploy this to staging (if you didn't already) and verify that the waterfall, tasks, variants work as expected (and that the legacy project dropdown button still redirects to spruce)

Checked the redirect still works, and all the regular task, version, build pages are working as expected, and ran a few restarts as well. No odd behavior or errors in Splunk

**Review by ablack12 — APPROVED** [review]

**Pull request merged**